frontend/src/containers/ReportStep/BoardMetricsChart/ClassificationChart/index.tsx

```diff
@@ -74,13 +212,74 @@ export const ClassificationChart = ({
   mappedData: ReportResponse[];
   dateRanges: string[];
 }) => {
+  const [isFirstIntoClassification, setIsFirstIntoClassification] = useState<boolean>(true);
+  const [isShowTimePeriodChart, setIsShowTimePeriodChart] = useState<boolean>(true);
+  const [canSwitchChart, setCanSwitchChart] = useState<boolean>(true);
+  const [rotate, setRotate] = useState<number>(0);
   const classificationRef = useRef<HTMLDivElement>(null);
-
-  const classificationData = extractClassificationData(classification, dateRanges, mappedData);
-  const classificationDataOption = classificationData && stackedBarOptionMapper(classificationData);
+  let classificationData;
+  let classificationDataOption;
+  if (isShowTimePeriodChart) {
+    classificationData = extractClassificationData(classification, dateRanges, mappedData);
+    classificationDataOption =
+      classificationData && stackedBarOptionMapper(classificationData, true, isFirstIntoClassification);
+  } else {
+    const chartType = checkClassificationChartType(classification, mappedData);
+    if (chartType === ClassificationChartType.Pie) {
+      classificationData = extractClassificationCardCountsPieData(classification, mappedData);
+      classificationDataOption = classificationData && pieOptionMapper(classificationData);
+    } else {
+      classificationData = extractClassificationCardCountsBarData(classification, mappedData);
+      classificationDataOption =
+        classificationData && stackedAreaOptionMapper(classificationData, true, isFirstIntoClassification);
+    }
+  }
   const isClassificationFinished =
     mappedData.flatMap((value) => value.classification).filter((it) => it?.name === classification)?.length ===
     dateRanges.length;
+  const isOnlyOneLegend = classificationDataOption.legend.data?.length === 0;
+
+  const transition = {
+    transform: `rotateY(${rotate}deg)`,
+  };
+
+  const maxRotateDeg = 90;
+  let id: number;
+  let start: number = 0;
+  function step(timestamp: number) {
```

Comment:
what is `setup`

Reply:
this is an animation. We hope it can flip how much deg, so the step is the step for every frame

Reply:
change the name to animationStep